### PR TITLE
Add support for SUSE systems

### PR DESCRIPTION
**Pull request by cryptomilk**
This PR adds support for SUSE systems.

@danimo

### CODE REVIEW

File: 46sshd/module-setup.sh [hunk 1]
```diff
@@ -62,7 +62,9 @@ install() {
     grep '^sshd:' /etc/passwd >> "$initdir/etc/passwd"
     grep '^sshd:' /etc/group  >> "$initdir/etc/group"
 
-    mkdir -p "$initdir/var/empty/sshd"
+    # Create privilege seperation directory
+    /usr/bin/install -d -m 755 "$initdir/var/empty/sshd"
+    /usr/bin/install -d -m 755 "$initdir/var/lib/empty"
```
Comment (gsauthof) [inline 46sshd/module-setup.sh]: I'm not a fan having an extra directory show up in the initramfs image. For example when debugging stuff one would have to remember which directory is actually supposed to be used.

What about making this conditional, e.g.:

    if [ -d /var/empty/ssh ]; then
        mkdir -p -m 755 "$initdir/var/empty/sshd"
    else
        mkdir -p -m 755 "$initdir/var/lib/empty"
    fi

File: 46sshd/module-setup.sh [hunk 2]
```diff
@@ -45,8 +45,8 @@ install() {
         return 1
     fi
 
-    mkdir -p -m 700 "$initdir/root/.ssh"
-    chmod 700 "$initdir/root"
+    /usr/bin/install -d -m 700 "$initdir/root"
+    /usr/bin/install -d -m 700 "$initdir/root/.ssh"
```
Comment (gsauthof) [inline 46sshd/module-setup.sh]: Another shorter alternative:

    mkdir -m 700 "$initdir/root"
    mkdir -m 700 "$initdir/root/.ssh"

File: 46sshd/module-setup.sh [hunk 3]
```diff
@@ -50,8 +50,9 @@ install() {
     /usr/bin/install -m 600 "$authorized_keys" \
             "$initdir/root/.ssh/authorized_keys"
 
-    inst_multiple /usr/sbin/sshd \
-        /etc/sysconfig/sshd
+    inst_simple /usr/sbin/sshd
+    inst_multiple -o /etc/sysconfig/ssh /etc/sysconfig/sshd
+
```
Comment (gsauthof) [inline 46sshd/module-setup.sh]: I think that this should fail if neither is available, e.g.:

    if [ -f /etc/sysconfig/sshd ]; then
        inst_simple /etc/sysconfig/sshd
    else
        inst_simple /etc/sysconfig/ssh
    fi

Comment (danimo) [reply 46sshd/module-setup.sh]: The lack of these files is perfectly legitimate, isn't it?

Comment (gsauthof) [reply 46sshd/module-setup.sh]: You are right. For example on Fedora, the file just contains comments. It's also already marked optional in the service file.

Thus, this hunk is fine, as is.

File: 46sshd/module-setup.sh (hunk 2 above)
Comment (cryptomilk) [reply 46sshd/module-setup.sh]: Fixed.

**Comment by cryptomilk** [comment]
Updated and pushed :-)

**Comment by gsauthof** [comment]
One last thing - what is the SUSE version you have tested this with?

**Comment by cryptomilk** [comment]
openSUSE Leap 15.1

**Pull request merged**